### DISCUSSION TITLE
Managing multiple GitHub repositories

### DISCUSSION THREAD

**Discussion by kiwiheretic** [community/community#64279] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have two different GitHub repositories which serve different purposes.  One is for work and the other is for myself.  However when I am switching between the two to say, work on a personal project, I have to log out of GitHub and then log back in with my personal project, find my phone to put in the verification code,  then go back to the command line to type in "gh auth login" and then finally copy and paste a browser code.  

As I often switch between personal and work projects is there a way to avoid doing this every time I switch between the two?  Has anyone else solved this problem without buying a second computer?

Thanks.

**Comment by 0xChintan** [comment]
You can use a neat trick! Instead of logging out, you can set up different profiles or SSH keys for your personal and work GitHub accounts on your computer. This way, you can easily switch between them without the hassle of logging in and out each time. It's like having different user accounts on the same computer. No need to buy a second computer!

**Comment by zainmujahid** [comment]
When switching between your work and personal GitHub repositories, you can streamline the process by implementing the following strategies:

#### 1- Multiple Git Configurations:
Utilize Git's ability to set up distinct configurations for different projects. This can be achieved by configuring separate user names and email addresses for your personal and work projects.
```
# Personal Configuration
git config --global user.name "Your Personal Name"
git config --global user.email "yourpersonal@email.com"

# Work Configuration
git config --global user.name "Your Work Name"
git config --global user.email "yourwork@email.com"
```

#### 2-  SSH Keys:
Set up SSH keys for both your personal and work GitHub accounts. This way, you can authenticate without needing to enter your password or verification code every time.

- Generate SSH keys: Follow GitHub's guide to [Generating a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).

- Add SSH keys to your accounts: Add the generated SSH keys to both your personal and work GitHub accounts.

#### 3-  Use HTTPS with Git Credentials Manager:
If you prefer using HTTPS instead of SSH, configure Git to use a credential manager. This way, you'll only need to enter your credentials once and they'll be cached.

- Enable credential manager: Run the following command to enable Git's built-in credential manager.

  ```
  git config --global credential.helper manager
  ```

#### 4-  GitHub CLI Personal Access Tokens:
The GitHub CLI (gh) allows you to use personal access tokens for authentication. Generate tokens for both your personal and work accounts and store them securely. You can then use the tokens in your command-line workflows.
```
# Configure personal token
gh auth login --with-token <your-personal-token>

# Configure work token
gh auth login --with-token <your-work-token>
```

By using these techniques, you can seamlessly switch between your personal and work GitHub repositories without repeatedly logging in or dealing with verification codes. These methods provide convenience while maintaining the security of your accounts.
